File: internal/extractor/scip/testdata/fixtures/typescript-express/src/repository.ts
// CampaignRepository is the data-access layer. The integration test
// asserts the walker reaches `findById` through the service.
export class CampaignRepository {
  private store = new Map<string, string>();

  findById(id: string): string | undefined {
    return this.store.get(id);
  }

  save(id: string, value: string): void {
    this.store.set(id, value);
  }
}

File: sources/extractor/internal/scip/testdata/fixtures/python-fastapi/app/repository.py
"""CampaignRepository is the data-access layer. The walker integration
test asserts that the path from the handler reaches `find_by_id`."""


class CampaignRepository:
    def __init__(self) -> None:
        self._store: dict[str, str] = {}

    def find_by_id(self, id: str) -> str | None:
        return self._store.get(id)

    def save(self, id: str, value: str) -> None:
        self._store[id] = value
